### PR TITLE
Added Crossword Puzzle Code in Python

### PR DESCRIPTION
**Pull request by arnab031**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Implement Crossword Puzzle in Python #2918 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Added Crossword Puzzle Code in python and update readme.md

### CODE REVIEW

File: .deepsource.toml [hunk 1]
```diff
@@ -6,25 +6,17 @@ enabled = true
...
   runtime_version = "3.x.x"
-  type_checker = "mypy"
 
 [[analyzers]]
 name = "go"
 enabled = true
 
   [analyzers.meta]
-  import_paths = ["github.com/TesseractCoding/NeoAlgo"]
+  import_paths = ["github.com/arnab031/NeoAlgo"]
```
Comment (HarshCasper) [inline .deepsource.toml:15]: Why did you change the `.deepsource.toml` after all?

File: .deepsource.toml [hunk 2]
```diff
@@ -6,25 +6,17 @@ enabled = true
...
 
   [analyzers.meta]
   environment = ["nodejs"]
-
-[[analyzers]]
-name = "java"
-enabled = true
-
-  [analyzers.meta]
-  runtime_version = "8"
```
Comment (HarshCasper) [inline .deepsource.toml:30]: Why did you remove this?

File: Python/cp/crossword_puzzle.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+Problem Link:https://www.hackerrank.com/challenges/crossword-puzzle/problem
+Steps:
+1. Check the cell to be '-' of first character of the word
+2. Check if that particular word can be filled in That place horizontally or vertically
+3. If the word can be filled , fill it!
+4. If it cann't be filled,Backtrack the previous word and fill it in other position
+5. Repeat the process until all the words are filled that is index == words.length.'''
+
+
+def solveCrossword(crossword0, words0, index0, wordfill0):
```
Comment (HarshCasper) [inline Python/cp/crossword_puzzle.py:11]: Add function docstring

File: Python/cp/crossword_puzzle.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                    removeVertically(i0, j0, crossword0, words0, index0, wordfill0)
+                # try to insert it horizontally
+                inserted = insertHorPoss(i0, j0, crossword0, words0, index0, wordfill0)
+                if inserted:
+                    solveCrossword(crossword0, words0, index0 + 1, wordfill0)
+                if inserted:
+                    removeHorizontally(i0, j0, crossword0, words0, index0, wordfill0)
+
+
+def removeHorizontally(i1, j1, crossword1, words1, index1, wordfill1):
```
Comment (HarshCasper) [inline Python/cp/crossword_puzzle.py:32]: Add function docstring

File: Python/cp/crossword_puzzle.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    for x1 in enumerate(words1[index1]):
+        if wordfill1[index1][x1]:
+            crossword1[i1][j1] = '-'
+        wordfill1[index1][x1] = False
+        j1 = j1+1
+
+
+def insertHorPoss(i2, j2, crossword2, words2, index2, wordfill2):
```
Comment (HarshCasper) [inline Python/cp/crossword_puzzle.py:40]: Add function docstring

File: Python/cp/crossword_puzzle.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                else:
+                    crossword2[i2][initial_j] = word[x2]
+                    wordfill2[index2][x2] = True
+                initial_j = initial_j+1
+            return True
+        j2 = j2+1
+    return False
+
+
+def insertVerPoss(i3, j3, crossword3, words3, index3, wordfill3):
```
Comment (HarshCasper) [inline Python/cp/crossword_puzzle.py:66]: Add function docstring

File: Python/cp/crossword_puzzle.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                else:
+                    crossword3[initial_i][j3] = word[x3]
+                    wordfill3[index3][x3] = True
+                initial_i = initial_i+1
+            return True
+        i3 = i3+1
+    return False
+
+
+def removeVertically(i4, j4, crossword4, words4, index4, wordfill4):
```
Comment (HarshCasper) [inline Python/cp/crossword_puzzle.py:92]: Add function docstring

File: Python/cp/crossword_puzzle.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+    for x4 in enumerate(words4[index4]):
+        if wordfill4[index4][x4]:
+            crossword4[i4][j4] = '-'
+        wordfill4[index4][x4] = False
+        i4 = i4+1
+
+
+def printCrossword(crossword5):
```
Comment (HarshCasper) [inline Python/cp/crossword_puzzle.py:100]: Add function docstring

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
No update

**Pull request closed without merging**